tap_integration_tests/tests/showcase.rs

```diff
@@ -778,6 +775,95 @@ async fn test_tap_manager_rav_timestamp_cuttoff(
+        http_request_size_limit,
+        http_response_size_limit,
+        http_max_concurrent_connections,
+    )
+    .await?;
+    let client =
+        HttpClientBuilder::default().build(format!("http://{}", gateway_addr.to_string()))?;
+
+    // This is the first part of the test, two batches of receipts are sent to the aggregator.
+    // The second batch has one receipt with the same timestamp as the latest receipt in the first batch.
+    // The first RAV will have the same timestamp as one receipt in the second batch.
+    // tap_aggregator should reject the second RAV request due to the repeated timestamp.
+    let requests = repeated_timestamp_request.await?;
+    let first_batch = &requests[0..receipt_threshold_1 as usize];
+    let second_batch = &requests[receipt_threshold_1 as usize..2 * receipt_threshold_1 as usize];
+
+    let receipts = first_batch
+        .iter()
+        .map(|(r, _)| r.clone())
+        .collect::<Vec<_>>();
+    let params = rpc_params!(&aggregate_server_api_version(), &receipts, None::<()>);
+    let first_rav_response: jsonrpsee_helpers::JsonRpcResponse<SignedRAV> =
+        client.request("aggregate_receipts", params).await?;
+
+    let receipts = second_batch
+        .iter()
+        .map(|(r, _)| r.clone())
+        .collect::<Vec<_>>();
+    let params = rpc_params!(
+        &aggregate_server_api_version(),
+        &receipts,
+        first_rav_response.data
+    );
+    let second_rav_response: Result<
+        jsonrpsee_helpers::JsonRpcResponse<SignedRAV>,
+        jsonrpsee::core::Error,
+    > = client.request("aggregate_receipts", params).await;
+    match second_rav_response {
+        Ok(_) => panic!("Should have failed RAV request"),
+        Err(_) => {}
+    }
+
+    // This is the second part of the test, two batches of receipts are sent to the aggregator.
+    // The second batch has one receipt with the timestamp = timestamp+1 of the latest receipt in the first batch.
+    // tap_aggregator should accept the second RAV request.
+    let requests = repeated_timestamp_incremented_by_one_request.await?;
+    let first_batch = &requests[0..receipt_threshold_1 as usize];
+    let second_batch = &requests[receipt_threshold_1 as usize..2 * receipt_threshold_1 as usize];
+
+    let receipts = first_batch
+        .iter()
+        .map(|(r, _)| r.clone())
+        .collect::<Vec<_>>();
+    let params = rpc_params!(&aggregate_server_api_version(), &receipts, None::<()>);
+    let first_rav_response: jsonrpsee_helpers::JsonRpcResponse<SignedRAV> =
+        client.request("aggregate_receipts", params).await?;
+
+    let receipts = second_batch
+        .iter()
+        .map(|(r, _)| r.clone())
+        .collect::<Vec<_>>();
+    let params = rpc_params!(
+        &aggregate_server_api_version(),
+        &receipts,
+        first_rav_response.data
+    );
+    let _second_rav_response: jsonrpsee_helpers::JsonRpcResponse<SignedRAV> =
+        client.request("aggregate_receipts", params).await?;
```

Comment:
You should also check the the RAV has the expected aggregate value, to make extra sure that no receipts got dropped near the previous RAV timestamp boundary.

Reply:
Added the check.